yurt/fabfile/add.py

```diff
@@ -108,18 +110,69 @@ def remote_server():
     recursive_file_modify("./templates.tmp", settings)
     with lcd("./templates.tmp"):
         for filepath, dest_template in TEMPLATE_TO_PROJECT_MAPPING.iteritems():
-            destination = dest_template.format(settings.get("project_name"),
+            destination = dest_template.format(vagrantfile_path.rstrip('/'),
+                                               settings.get("project_name"),
                                                settings.get("abbrev_env"),
                                                settings.get("env"))
             local("mv {0} {1}".format(filepath, destination))
     local("rm -rf ./templates.tmp")
 
 
 @main.command()
-def vault():
+@click.option('--vault', is_flag=True, help="Uses vault for git keys")
+def add_settings(vault):
+    """
+    Adds `fabric_settings.py` to this directory
+    :return: Void
+    """
+    public_key, private_key = generate_ssh_keypair()
+    settings = {
+        'git_public_key': public_key,
+        'git_private_key': private_key,
+        'vagrant': {
+            'db_pw': generate_printable_string(15, False),
+            'secret_key': generate_printable_string(40)
+        }
+    }
+    if vault:
+        vagrantfile_dir = find_vagrantfile_dir()
```

Comment:
to keep this method shorter I wonder if the code under `if vault:` should be put in a separate method